aten/src/ATen/native/cpu/ReduceOpsKernel.cpp

```diff
@@ -136,15 +136,34 @@ static void logcumsumexp_cpu_kernel(Tensor& result, const Tensor& self, int64_t
   });
 }
 
-static void mean_kernel_impl(TensorIterator& iter) {
-  AT_DISPATCH_ALL_TYPES_AND_COMPLEX(iter.dtype(), "mean_cpu", [&] {
-    scalar_t factor = scalar_t(iter.num_output_elements()) / scalar_t(iter.numel());
-    binary_kernel_reduce(
+template <typename scalar_t, typename acc_t=scalar_t, typename out_t=scalar_t>
+void mean_kernel_impl(TensorIterator& iter) {
```

Comment:
FYI `MeanOps` isn't actually used on CPU, see `TORCH_IMPL_FUNC(mean_out)`
https://github.com/pytorch/pytorch/blob/364d92f9b6864ce284fa13519c7ca5c87460e477/aten/src/ATen/native/ReduceOps.cpp#L1318

Reply:
I guess we'd have to make `binary_kernel_reduce_vec` work with `mean` first. Then, we don't have to invoke `sum_out` + `div_` there.

Reply:
Currently, it may be better to call ```mean_stub``` path for bfloat16 and half, I met an accuracy issue when comparing ```nn.AdaptiveAvgPool2d```(calling mean) with inductor(using fused kernel, sum+div):
```
import torch
import torch._dynamo

class Model(torch.nn.Module):
    def __init__(self):
        super(Model, self).__init__()
        self.avgpool = torch.nn.AdaptiveAvgPool2d((1, 1))

    def forward(self, x):
        x = self.avgpool(x)
        return x

model = Model().eval()
opt_model = torch._dynamo.optimize('inductor')(model)

batch_size = 4
x = torch.randn(batch_size, 512, 7, 7).to(dtype=torch.bfloat16)
ref = model(x)
with torch.no_grad():
    for i in range(3):
        out = opt_model(x)

print((ref - out).max())
tol = 0.001
print(ref-out)
print(torch.allclose(ref, out, atol=tol, rtol=tol))
````

before this PR, this case always failed, the eager model seems to have a big difference with inductor result, @mingfeima has the plan to support the reduction vectorization kernel.